media/js/base/core-datalayer.js

```diff
@@ -0,0 +1,34 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// create namespace
+if (typeof Mozilla == 'undefined') {
+    var Mozilla = {};
+}
+
+/**
+* Utility class for core dataLayer object to track contextual user and page data
+*/
+
+(function() {
+    var dataLayer = window.dataLayer = window.dataLayer || [];
+    var Analytics = {};
+
+    // Reurns page ID used in Event Category for GA events tracked on page
+    Analytics.getPageId = function() {
+        var pageId = $('html').attr('data-gtm-page-id');
+
+        return pageId ? pageId : document.location.pathname.replace(/\/[^\/]*/, '');
+    };
+
+    // Push core datalayer object
+    dataLayer.push({
+        'event': 'core-dataLayer-loaded',
+        'pageId': Analytics.getPageId()
```

Comment:
As we're writing tests for this file, and will be expanding on the functions defined here in the future, I think it makes sense to separate out when the functions are being run. Currently this is executing before the tests are run, which while isn't breaking anything, probably isn't ideal - and could lead to unexpected results when we expand on things in the future. 

I would suggest creating a file called `core-datalayer-init.js` which is not included in the test runner, similar to what we do with global.js here: https://github.com/mozilla/bedrock/blob/master/media/js/base/global-init.js